src/Stratis.Bitcoin.Features.Miner/Staking/PosMinting.cs

```diff
@@ -434,7 +434,7 @@ internal async Task<List<UtxoStakeDescription>> GetUtxoStakeDescriptionsAsync(Wa
                 .ToList();
 
             FetchCoinsResponse fetchedCoinSet = await this.coinView.FetchCoinsAsync(stakableUtxos.Select(t => t.Transaction.Id).Distinct().ToArray(), cancellationToken).ConfigureAwait(false);
-            Dictionary<uint256, UnspentOutputs> utxoByTransaction = fetchedCoinSet.UnspentOutputs.ToDictionary(utxo => utxo.TransactionId, utxo => utxo);
+            Dictionary<uint256, UnspentOutputs> utxoByTransaction = fetchedCoinSet.UnspentOutputs.Where(utxo => utxo != null).ToDictionary(utxo => utxo.TransactionId, utxo => utxo);
```

Comment:
I suspect actually that this bug means the wallet is out of sync (it can happen if its catching up or after a regorg) so maybe the right way to fix it is abort and return back up tot eh main loop WDYT?

Reply:
I hear what you are saying but do we really want to make that assumption and potentially introduce other issues caused by the assumption? If its valid to have null outputs then one should simply ensure that code further along recognizes that fact. 

Also we should not have to write code all over the place to deal with re-orgs. The right fix for what you are describing is for the consensus manager to pro-actively invoke methods (e.g. `DeleteBlocks`) **on features to keep them in sync** when re-orgs occur. Individual features should not have to deal with the nitty-gritty of re-orgs. Each feature should simply consume or delete (or hide and then lazily delete) blocks when instructed to do so.